operator/deploy/deploy.go

```diff
@@ -232,6 +222,38 @@ func (o *operator) deployment() *appsv1.Deployment {
 			},
 		},
 	}
+	if role == "master" {
+		deployment.Spec.Template.Spec.Volumes = []v1.Volume{
+			{
+				Name: "pullsecret-tokens",
+				VolumeSource: v1.VolumeSource{
+					Secret: &v1.SecretVolumeSource{
+						SecretName: "pullsecret-tokens",
+					},
+				},
+			},
+		}
+		deployment.Spec.Template.Spec.NodeSelector = map[string]string{"node-role.kubernetes.io/master": ""}
+		deployment.Spec.Template.Spec.Tolerations = []v1.Toleration{
```

Comment:
you are not setting this stuff for worker? I assume it defaults to non-master?

Reply:
I will setup NodeSelector explicitly for the worker. But generally if the deployment doesn't have specific toleration set, then it get scheduled on the nodes without that toleration. This ensures worker don't get scheduled on a master